core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataReader.java

```diff
@@ -174,7 +174,12 @@ private DataBuffer readChunkInternal() throws IOException {
       // finish reading all the data, close gRPC stream earlier to release resources
       // potentially improve machine learning millions of small file read throughput
       LOG.debug("Finished reading all data, close gRPC stream");
-      close();
+      try {
+        close();
+      } catch (Throwable t) {
+        buffer.release();
```

Comment:
We had the buffer already. Why not just return it to the caller and then the caller owns releasing it after consumption.